packages/mco/components/disaster-recovery/app-failover-relocate/failover-relocate-modal-body.tsx

```diff
@@ -0,0 +1,289 @@
+import { PeerClusterStatus } from './peer-cluster-status';
+import {
+  FailoverAndRelocateState,
+  FailoverAndRelocateAction,
+  FailoverAndRelocateType,
+  DRPolicyControlState,
+  ACTION_TYPE,
+  ErrorMessage,
+  ModalFooterStatus,
+} from './reducer';
+import { SubscriptionGroupSelector } from './subscription-group-selector';
+import { TargetClusterSelector } from './target-cluster-selector';
+import './failover-relocate-modal-body.scss';
+
+export const findErrorMessage = (
+  errorMessage: ErrorMessage,
+  includeUpdateError: boolean = true
+) =>
+  [
+    errorMessage.drPolicyControlStateErrorMessage,
+    errorMessage.failoverAndRelocateActionErrorMessage,
+    errorMessage.targetClusterErrorMessage,
+    errorMessage.subscriptionGroupErrorMessage,
+    errorMessage.peerStatusErrorMessage,
+    includeUpdateError && errorMessage.failoverAndRelocateActionErrorMessage,
+  ]
+    .filter(Boolean)
+    .find((errorMessage) => errorMessage);
+
+const getDRPolicyControlState = (
+  drPlacementControls: DRPlacementControlKind[],
+  subscriptionMap: SubcsriptionMap
+): DRPolicyControlState[] => {
+  return drPlacementControls?.reduce(
+    (acc, drPlacementControl) =>
+      isObjectRefMatching(
+        drPlacementControl?.spec?.placementRef,
+        Object.keys(subscriptionMap)
+      )
+        ? [
+            ...acc,
+            {
+              drPolicyControl: drPlacementControl,
+              subscriptions:
+                subscriptionMap?.[drPlacementControl?.spec?.placementRef?.name],
+            },
+          ]
+        : acc,
+    []
+  );
+};
+
+const resources = (namespace: string) => ({
+  subscriptions: {
+    kind: referenceForModel(ACMSubscriptionModel),
+    namespaced: true,
+    namespace,
+    isList: true,
+  },
+  placementRules: {
+    kind: referenceForModel(ACMPlacementRuleModel),
+    namespaced: true,
+    namespace,
+    isList: true,
+  },
+  drPlacementControls: {
+    kind: referenceForModel(DRPlacementControlModel),
+    namespaced: true,
+    namespace,
+    isList: true,
+  },
+});
+
+const getMessageStatus = (message: string, variant: AlertVariant) => (
+  <Flex>
+    <FlexItem fullWidth={{ default: 'fullWidth' }}>
+      <Alert
+        className="mco-dr-action-body__alert"
+        title={message}
+        variant={variant}
+        isInline
+      />
+    </FlexItem>
+  </Flex>
+);
+
+export const FailoverRelocateModalBody: React.FC<FailoverRelocateModalBodyProps> =
+  (props) => {
+    const { application, action, state, dispatch } = props;
+    const { t } = useCustomTranslation();
+
+    const response = useK8sWatchResources<DRActionWatchResourceType>(
+      resources(application?.metadata?.namespace)
+    );
+    const memoizedPlacementRule = useDeepCompareMemoize(
+      response.placementRules
+    );
+    const memoizedSubscriptions = useDeepCompareMemoize(response.subscriptions);
+    const memoizedDRPlacementControls = useDeepCompareMemoize(
+      response.drPlacementControls
+    );
```

Comment:
```suggestion
    const memoizedSubscriptions = useDeepCompareMemoize(response.subscriptions);
    const memoizedDRPlacementControls = useDeepCompareMemoize(
      response.drPlacementControls
    );
```
You prolly would want to pass the second arg as `true` to compare string values.